pmlc/conversion/tile_to_pxa/tests/compare.mlir

```diff
@@ -0,0 +1,201 @@
+// CHECK: cmpf "olt"
+  %4 = "eltwise.cmp_le"(%i32, %0) {type = !eltwise.fp32} : (!i32, !fp32) -> !bool
+// CHECK: pxa.parallel_for
+// CHECK: sitofp
+// CHECK: cmpf "ole"
+  %5 = "eltwise.cmp_gt"(%i32, %0) {type = !eltwise.fp32} : (!i32, !fp32) -> !bool
+// CHECK: pxa.parallel_for
+// CHECK: sitofp
+// CHECK: cmpf "ogt"
+  %6 = "eltwise.cmp_ge"(%i32, %0) {type = !eltwise.fp32} : (!i32, !fp32) -> !bool
+// CHECK: pxa.parallel_for
+// CHECK: sitofp
+// CHECK: cmpf "oge"
+  return %1, %2, %3, %4, %5, %6 : !bool, !bool, !bool, !bool, !bool, !bool
+}
+
+
+// -----
+
+!i16 = type tensor<!eltwise.i16>
+!bool = type tensor<!eltwise.bool>
+
+// CHECK-LABEL: func @cmp_i16_i16
+func @cmp_i16_i16(%i16: !i16) -> !bool {
+  %c0 = "eltwise.sconst"() {value = 0 : i64} : () -> !i16
+  %0 = "eltwise.cmp_lt"(%i16, %c0) {type = !eltwise.fp32} : (!i16, !i16) -> !bool
+  // CHECK: pxa.parallel_for
+  // CHECK: cmpi "slt"
+  return %0 : !bool
+}
+
+// -----
+
+!i16 = type tensor<!eltwise.i16>
+!u16 = type tensor<!eltwise.u16>
+!bool = type tensor<!eltwise.bool>
+
+// CHECK-LABEL: func @cmp_i16_u16
+func @cmp_i16_u16(%i16: !i16) -> !bool {
+  %c0 = "eltwise.sconst"() {value = 0 : i64} : () -> !u16
+  %0 = "eltwise.cmp_lt"(%i16, %c0) {type = !eltwise.fp32} : (!i16, !u16) -> !bool
+  // CHECK: pxa.parallel_for
+  // CHECK: cmpi "ult"
+  return %0 : !bool
+}
+
+// -----
+
+!i16 = type tensor<!eltwise.i16>
+!i32 = type tensor<!eltwise.i32>
+!bool = type tensor<!eltwise.bool>
+
+// CHECK-LABEL: func @cmp_i16_i32
+func @cmp_i16_i32(%i16: !i16) -> !bool {
+  %c0 = "eltwise.sconst"() {value = 0 : i64} : () -> !i32
+  %0 = "eltwise.cmp_lt"(%i16, %c0) {type = !eltwise.fp32} : (!i16, !i32) -> !bool
+  // CHECK: pxa.parallel_for
+  // CHECK: sexti
+  // CHECK: cmpi "slt"
+  return %0 : !bool
+}
+
+// -----
+
+!i16 = type tensor<!eltwise.i16>
+!u32 = type tensor<!eltwise.u32>
+!bool = type tensor<!eltwise.bool>
+
+// CHECK-LABEL: func @cmp_i16_u32
+func @cmp_i16_u32(%i16: !i16) -> !bool {
+  %c0 = "eltwise.sconst"() {value = 0 : i64} : () -> !u32
+  %0 = "eltwise.cmp_lt"(%i16, %c0) {type = !eltwise.fp32} : (!i16, !u32) -> !bool
+  // CHECK: pxa.parallel_for
+  // CHECK: zexti
+  // CHECK: cmpi "ult"
+  return %0 : !bool
+}
+
+// -----
+
+!u16 = type tensor<!eltwise.u16>
+!bool = type tensor<!eltwise.bool>
+
+// CHECK-LABEL: func @cmp_u16_u16
+func @cmp_u16_u16(%u16: !u16) -> !bool {
+  %c0 = "eltwise.sconst"() {value = 0 : i64} : () -> !u16
+  %0 = "eltwise.cmp_lt"(%u16, %c0) {type = !eltwise.fp32} : (!u16, !u16) -> !bool
+  // CHECK: pxa.parallel_for
+  // CHECK: cmpi "ult"
+  return %0 : !bool
+}
+
+// -----
+
+!u16 = type tensor<!eltwise.u16>
+!i32 = type tensor<!eltwise.i32>
+!bool = type tensor<!eltwise.bool>
+
+// CHECK-LABEL: func @cmp_u16_i32
+func @cmp_u16_i32(%u16: !u16) -> !bool {
+  %c0 = "eltwise.sconst"() {value = 0 : i64} : () -> !i32
+  %0 = "eltwise.cmp_lt"(%u16, %c0) {type = !eltwise.fp32} : (!u16, !i32) -> !bool
+  // CHECK: pxa.parallel_for
+  // CHECK: sexti
```

Comment:
This should be definitely be a  zexti, if I do uint16_t(0x8000) < int32_t(0), it should return false.   In general, I think the extension follows the signedness of the input type.